### PR TITLE
decision making statements / it-academy / JC1-117-22 / Dzmitry Krasiuk / Done all tasks 

### PR DESCRIPTION
**Pull request by Dzmitry-Highlander**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task07.java [hunk 1]
```diff
@@ -36,8 +36,8 @@ public static void main(String[] args) {
      * @return <code>false</code>/<code>true</code>
      */
     static boolean isLeapYear(int year) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return false;
+        boolean isYearLeap = year % 4 == 0;
+
+        return isYearLeap;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task07.java:41]: WRONG

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task04.java [hunk 2]
```diff
@@ -29,8 +29,32 @@ public static void main(String[] args) {
...
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return null;
+        String dayOfTheWeek = "";
+
+        switch (dayOfWeekNumber) {
+            case 1:
+                dayOfTheWeek = "MONDAY";
+                break;
+            case 2:
+                dayOfTheWeek = "THURSDAY";
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task04.java:39]: WRONG

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task10.java [hunk 3]
```diff
@@ -35,8 +35,18 @@ public static void main(String[] args) {
...
+            firstToFuel = 1;
+        } else if (secondFuelReserves < firstFuelReserves && secondFuelReserves < thirdFuelReserves) {
+            firstToFuel = 2;
+        } else if (thirdFuelReserves < firstFuelReserves && thirdFuelReserves < secondFuelReserves) {
+            firstToFuel = 3;
+        } else {
+            firstToFuel = 1;
+        }
+
+        return firstToFuel;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task10.java:50]: WRONG

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task11.java [hunk 4]
```diff
@@ -38,8 +38,20 @@ public static void main(String[] args) {
      * @return <code>latin</code>/<code>cyrillic</code>/<code>digit</code>/<code>undefined</code>
      */
     static String getSymbolType(char symbol) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return null;
+        String unicodeChart;
+
+        if ((int) symbol >= 65529 && (int) symbol <= 65333) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task11.java:43]: "Code smell: Magic Number
Solution: Named Constants"

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task11.java [hunk 5]
```diff
@@ -38,8 +38,20 @@ public static void main(String[] args) {
...
+            unicodeChart = "special";
+        } else if ((int) symbol >= 65 && (int) symbol <= 90 || (int) symbol >= 97 && (int) symbol <= 122) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task11.java:45]: "Code smell: Magic Number
Solution: Named Constants"

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task11.java [hunk 6]
```diff
@@ -38,8 +38,20 @@ public static void main(String[] args) {
...
+            unicodeChart = "latin";
+        } else if ((int) symbol >= 1024 && (int) symbol <= 1273) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task11.java:47]: "Code smell: Magic Number
Solution: Named Constants"

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task11.java [hunk 7]
```diff
@@ -38,8 +38,20 @@ public static void main(String[] args) {
...
+            unicodeChart = "cyrillic";
+        } else if ((int) symbol >= 48 && (int) symbol <= 57) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task11.java:49]: "Code smell: Magic Number
Solution: Named Constants"

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task13.java [hunk 8]
```diff
@@ -32,8 +32,18 @@ public static void main(String[] args) {
...
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return 0;
+        int porchNumber = 1;
+        int apartmentsPerPorch = numberFloors * numberApartmentsPerFloor;
+        double floorNumber;
+
+        while (apartmentsPerPorch < apartmentNumber) {
+            porchNumber++;
+            apartmentsPerPorch += numberFloors * numberApartmentsPerFloor;
+        }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task13.java:42]: Need to add solution with `if`

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task14.java [hunk 9]
```diff
@@ -42,8 +42,43 @@ public static void main(String[] args) {
...
+        int chinaCalendarYear = 1;
+        String nameOfYear = "";
+
+        while (counter < year) {
+            counter++;
+            chinaCalendarYear++;
+            if (chinaCalendarYear > 12) {
+                chinaCalendarYear = 1;
+            }
+        }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task14.java:55]: Best way: use `%` operator

**Review by rakovets — CHANGES_REQUESTED** [review]
Need to fix

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task08.java [hunk 10]
```diff
@@ -38,8 +40,18 @@ public static void main(String[] args) {
      * @return итоговый урон по данному типу моба
      */
     static int getTotalDamage(int damage, String typeMob, boolean hasHolyAttribute) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return 0;
+        int totalDamage = 0;
+
+        if (hasHolyAttribute) {
```
[the remark this answers is not in the document]
Comment (Dzmitry-Highlander) [reply src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task08.java:43]: done

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task02.java [hunk 11]
```diff
@@ -35,8 +35,25 @@ public static void main(String[] args) {
...
     static String getGreetingByHour(int hour) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return null;
+        String helloWords;
+        int dayStart = 0;
+        int sunrise = 6;
+        int midnight = 24;
+        int midday = 12;
+        int sunset = 18;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task02.java:43]: use CONSTANTS

**Review by rakovets — APPROVED** [review]

**Pull request closed without merging**